nengo_gui/static/status_bar.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Simple JavaScript status bar which is used at the bottom of the net graph
+ * window.
+ *
+ * Creates a global instance Nengo.status_bar inside the netgraph which can be
+ * accessed using the following functions:
+ * - Nengo.status_bar.set_caption()
+ * - Nengo.status_bar.get_caption()
+ * - Nengo.status_bar.hide()
+ * - Nengo.status_bar.show()
+ *
+ * @author Andreas Stöckel
+ * @date November 2017
+ */
+
+Nengo.StatusBar = (function () {
+    "use strict";
+
+    /**
+     * StatusBar constructor. Instantiates the DOM for the status bar and
+     * attaches it to the given parent DOM element.
+     *
+     * @param parent is the DOM node the StatusBar should be attached to.
+     */
+    var StatusBar = function(parent) {
+        "use strict";
+
+        // Create the component by manually assembling the DOM
+        this.status_bar_div = document.createElement('div');
+        this.status_bar_div.classList.add('status_bar');
+
+        // Timeout for updating the caption
+        this.caption_update_timeout = null;
+
+        // Append the component to the parent container
```

Comment:
This comment doesn't really say anything, that the next line doesn't say already. I would remove it.

Reply:
Removed last comment and reworded first comment